d3dcore/include/d3dcore/d3d_info.h

```diff
@@ -0,0 +1,253 @@
+#pragma once
+
+#define INITGUID
+#include "dxdiag.h"
+#include <Windows.h>
+
+#include "d3d_utils.h"
+
+#include <string>
+#include <memory>
+#include <assert.h>
+#include <iostream>
+#include <string>
+#include <any>
+
+
+//-----------------------------------------------------------------------------
+// Defines, and constants
+//-----------------------------------------------------------------------------
+#define SAFE_DELETE(p)       { if(p) { delete (p);     (p)=NULL; } }
+#define SAFE_DELETE_ARRAY(p) { if(p) { delete[] (p);   (p)=NULL; } }
+#define SAFE_RELEASE(p)      { if(p) { (p)->Release(); (p)=NULL; } }
+#define SAFE_BSTR_FREE(x)    if(x) { SysFreeString( x ); x = NULL; }
+#define EXPAND(x)            x, sizeof(x)/sizeof(TCHAR)
+
+
+
+namespace d3dcore {
+
+	/*enum class SDKVersion {
```

Comment:
Do not leave commented code in production, version control exists exactly for that purposes